packages/rpc-core/src/rpc-methods/__tests__/get-program-accounts-test.ts

```diff
@@ -0,0 +1,1213 @@
+                'AmtpVzo6H6qQCP9dH9wfu5hfa8kKaAFpTJ4aamPYR6V6' as Base58EncodedAddress<'AmtpVzo6H6qQCP9dH9wfu5hfa8kKaAFpTJ4aamPYR6V6'>;
+
+            const accountInfo = await rpc.getProgramAccounts(program).send();
+
+            expect(accountInfo).toHaveLength(2);
+            expect(accountInfo[0]).toMatchObject({
+                account: expect.objectContaining({
+                    data: expect.any(String),
+                    executable: expect.any(Boolean),
+                    lamports: expect.any(BigInt),
+                    owner: expect.any(String),
+                    rentEpoch: expect.any(BigInt),
+                    space: expect.any(BigInt),
+                }),
+                pubkey: expect.any(String),
+            });
+            expect(accountInfo[1]).toMatchObject({
+                account: expect.objectContaining({
+                    data: expect.any(String),
+                    executable: expect.any(Boolean),
+                    lamports: expect.any(BigInt),
+                    owner: expect.any(String),
+                    rentEpoch: expect.any(BigInt),
+                    space: expect.any(BigInt),
+                }),
+                pubkey: expect.any(String),
+            });
+        });
+    });
+
+    describe('when called with base58 encoding', () => {
+        describe('when called with withContext: false', () => {
+            it('returns top-level account info with annotated base58 encoding', async () => {
+                expect.assertions(1);
+                // See scripts/fixtures/gpa1.json
+                // data is 'test data'
+                const program =
+                    'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj' as Base58EncodedAddress<'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj'>;
+
+                const accountInfo = await rpc
+                    .getProgramAccounts(program, {
+                        encoding: 'base58',
+                        withContext: false,
+                    })
+                    .send();
+
+                expect(accountInfo[0].account.data).toStrictEqual(['2Uw1bpnsXxu3e', 'base58']);
+            });
+        });
+
+        describe('when called with withContext: true', () => {
+            it('returns RPC Response with account info with annotated base58 encoding', async () => {
+                expect.assertions(1);
+                // See scripts/fixtures/gpa1.json
+                // data is 'test data'
+                const program =
+                    'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj' as Base58EncodedAddress<'DXngmJfjurhnAwbMPgpUGPH6qNvetCKRJ6PiD4ag4PTj'>;
+
+                const accountInfo = await rpc
+                    .getProgramAccounts(program, {
+                        encoding: 'base58',
+                        withContext: true,
+                    })
+                    .send();
+
+                expect(accountInfo.value[0].account.data).toStrictEqual(['2Uw1bpnsXxu3e', 'base58']);
```

Comment:
Seems a little unnecessary, but should we check the whole response to ensure it matches `RpcResponse`? ie. add the context bit to the `expect(..)`

Reply:
It's probably worth checking this somewhere, and since this is the first `withContext: true` test it's probably the place to do it

One thing I will point out is that these tests that traverse the data like `accountInfo.value[0].account.data` are actually really valuable for the type checking. Eg if you mess up the typing like when I was trying to do the conditional types, the tests that use `toMatchObject` etc. won't fail because the actual return data is correct. But `accountInfo.value` will error because it doesn't know what the type is. I suspect we have some RPC methods that we've only tested using object matchers that we're not really testing the typing of.

Reply:
What about if you are expecting:

```javascript
{
  context: number,
  value: [
    { account: { data: string }}
  ]
}
```

and you get:

```javascript
{
  foo: string,
  value: [
    { account: { data: string }}
  ]
}
```

Wouldn't `accountInfo.value[0].account.data` still work?

Reply:
Yep you have to test `context` for that (or use `toMatchObject` and check both fields)

I've added `expect(accountInfo.context.slot).toEqual(expect.any(BigInt));` to this test which covers it for this case

The rest of the `account` structure is IMO adequately tested elsewhere, but the RPC bit is only tested by these tests that have `withContext: true`